### DISCUSSION TITLE
Changes to Actor.anchor are only applied to the graphics component

### DISCUSSION THREAD

**Discussion by VaughanRSanders** [excaliburjs/Excalibur#3177] [Q&A] [answered]
Changing the anchor of the Actor seem to intentionally only change the graphics component 
```
private _handleAnchorChange(v: Vector) {
    if (this.graphics) {
      this.graphics.anchor = v;
    }
  }
```

This leads to the Actor having a physics component that is offset from the graphics - you can see in the screenshot the yellow box is the graphics bounds and the blue is the physics.  The anchor for this Actor is `vec(0.5, 1)` and can be seen as the yellow dot.
![Screenshot from 2024-08-24 11-07-54](https://github.com/user-attachments/assets/04594eb4-d2c8-4775-8c55-91780b853562)

Is there a way / should I be manually updating the physics bounds to match the graphics?

**Comment by Autsider666** [comment, marked as answer]
The easiest way to apply a new anchor to the collider of an actor outside of the constructor seems to be to create a new one using `actor.collider.set(Shape.Box(width, height, newAnchor))`.

You're running into this issue because the collider doesn't have any knowledge of the anchor of its entity.
It looks like the collider [gets the anchor in the actor constructor](https://github.com/excaliburjs/Excalibur/blob/main/src/engine/Actor.ts#L651), but the [`Box` function only uses it to create the initial points of the `PolygonCollider`](https://github.com/excaliburjs/Excalibur/blob/main/src/engine/Collision/Colliders/Shape.ts#L20).

**Reply by Autsider666** [reply]
@eonarheim would it be possible to extract [this logic](https://github.com/excaliburjs/Excalibur/blob/main/src/engine/Collision/Colliders/Shape.ts#L22) from `Shape.Box` so it can be used to generate net points for an existing `PolygonCollider` in case the anchor of an actor gets changed? Running `_calculateTransformation` afterwards seems to be enough, but colliders are still mostly magic to me.

**Reply by eonarheim** [reply]
Thanks, agreed. I think the easiest way around this is to manage your collider anchor and graphics anchor separately.

Alternatively you could look at setting specific pixel `offset` on your collider and graphics if you know exact offsets

```typescript
actor.graphics.offset = ex.vec(0, -100); // 100 pixels up

const collider = ex.Shape.Box(100, 100, ex.Vector.Half , ex.vec(0, -100)); // 100 pixels up
actor.collider.set(collider);
```

**Reply by eonarheim** [reply]
> @eonarheim would it be possible to extract [this logic](https://github.com/excaliburjs/Excalibur/blob/main/src/engine/Collision/Colliders/Shape.ts#L22) from `Shape.Box` so it can be used to generate net points for an existing `PolygonCollider` in case the anchor of an actor gets changed? Running `_calculateTransformation` afterwards seems to be enough, but colliders are still mostly magic to me.

Totally, I was thinking we could probably do a convenience top level `.offset` on the ColliderComponent itself

**Reply by VaughanRSanders** [reply]
Thank you both for the reply.  My thinking around using `anchor` rather than `offset` was I would like to be able to tell the actor to `moveTo` specific world positions and have the center of the feet arrive at that point - this seems like the correct way to setup that from a `transform` point of view.

I will manually set the collider :pray:
